api/sendfild/service/service.go

```diff
@@ -0,0 +1,713 @@
+package service
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math"
+	"math/big"
+	"net"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/ipfs/go-cid"
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/textileio/go-threads/util"
+	"github.com/textileio/powergate/v2/lotus"
+	pb "github.com/textileio/textile/v2/api/sendfild/pb"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+const (
+	collectionName = "sendfil"
+)
+
+var _ pb.SendFilServiceServer = (*Service)(nil)
+
+var log = logging.Logger("sendfil")
+
+type msgCid struct {
+	Cid       string    `bson:"cid"`
+	CreatedAt time.Time `bson:"created_at"`
+}
```

Comment:
As a future proof thing, we support tracking multiple cids for a single message. This feature doesn't exist yet, but we could do something like: If a message is stuck in the mpool because the max gas fees are too low (or any other reason I'm not yet aware of), we could write code that detects that and automatically replaces the message with one that uses better max gas fee limits, etc. This new message would have an updated cid, and that cid would be added to the list of cids we track on a `txn` record. We then have the ability to look up a `txn` for any cid that has been associated with it over time.